File: skillhunter/skillhunter/Skriptum.vb
Public MustInherit Class Skriptum
    Public Enum ObjectType
        SCRIPT
        MISSION_TEMPLATE
        PRESENTATION
        GAME_MENU
        TROOP
        ITEM
        GAME_STRING
        SIMPLE_TRIGGER
        TRIGGER
        INFO_PAGE
        MESH
        MUSIC
        QUEST
        SOUND
        SCENE_PROP
        TABLEAU_MATERIAL
        MAP_ICON
        DIALOG
        FACTION
        ANIMATION
        PARTY_TEMPLATE
        PARTY
        SKILL
        POST_FX
        SKIN
        PARTICLE_SYSTEM
        SCENE
    End Enum

    Public Shared ReadOnly Prefixes As String() = {"script", "mt", "prsnt", "menu", "trp", "itm", "str", "simple_trigger", "trigger", "ip", "mesh", "track", "qst", "snd", "spr", "tableau", "icon",
                                                    "dialog", "fac", "anim", "pt", "p", "skl", "pfx", "skin", "psys", "scn"}

    Public ReadOnly Property ID As String
    Public ReadOnly Property ObjectTyp As ObjectType

    Public Sub New(sIdName As String, type As ObjectType)

        sIdName = sIdName.Trim()
        If sIdName.StartsWith(Prefix) Then
            sIdName = sIdName.Substring(Prefix.Length)
        End If

        ObjectTyp = type
        ID = sIdName
    End Sub

    Public ReadOnly Property Typ As Integer
        Get
            Return ObjectTyp
        End Get
    End Property

    Public ReadOnly Property Prefix As String
        Get
            Return Prefixes(ObjectTyp) + "_"c
        End Get
    End Property

End Class

File: MB_Decompiler_Library/Objects/Skriptum.cs
namespace MB_Decompiler_Library.Objects
{
    public abstract class Skriptum
    {
        public static readonly string[] Prefixes = {"script", "mt", "prsnt", "menu", "trp", "itm", "str", "simple_trigger", "trigger", "ip", "mesh", "track", "qst", "snd", "spr", "tableau", "icon",
                                                    "dialog", "fac", "anim", "pt", "p", "skl", "pfx", "skin", "psys", "scn"};

        public enum ObjectType
        {
            Script,
            MissionTemplate,
            Presentation,
            GameMenu,
            Troop,
            Item,
            GameString,
            SimpleTrigger,
            Trigger,
            InfoPage,
            Mesh,
            Music,
            Quest,
            Sound,
            SceneProp,
            TableauMaterial,
            MapIcon,
            Dialog,
            Faction,
            Animation,
            PartyTemplate,
            Party,
            Skill,
            PostFX,
            Skin,
            ParticleSystem,
            Scene
        }

        public string ID { get; }

        public ObjectType ObjectTyp { get; }

        public int Typ { get { return (int)ObjectTyp; } }

        public string Prefix { get { return Prefixes[Typ] + '_'; } }

        public Skriptum(string sIdName, ObjectType type)
        {
            ObjectTyp = type;

            sIdName = sIdName.Trim();
            if (sIdName.StartsWith(Prefix))
                sIdName = sIdName.Substring(Prefix.Length);
            ID = sIdName;
        }

    }
}
